net: wifi: shell: Fix connect command SSID parameter

loicpoulain opened the pull request:
The new shell subsystem takes care of removing quote ('"') from
command arguments and produces a standard string.
So now the SSID string starts at &argv[1][0] (instead of &argv[1][1]).
Fix this.

In the same time, remove SSID length parameter which can be find with
strlen().

Signed-off-by: lpoulain <lpoulain@lpoulain-ThinkPad-T470p>

jukkar commented:
@GAnthony I wonder if this is fixing the issues you had earlier with wifi shell?

tbursztyka commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -149,22 +149,15 @@ static int cmd_wifi_connect(const struct shell *shell, size_t argc,
-
-	argv[1][cnx_params.ssid_length + 1] = '\0';
+	cnx_params.ssid = argv[1];
+	cnx_params.ssid_length = strlen(argv[1]);
```
The WiFi specifies SSID as being a byte array, not a string. Thus why you could have anything there that would make this strlen() not going right. That said, nor this shell module is able to get escaped input (tough it could be implemented) nor the below device drivers below are handling the ssid properly (ti, atmel... they all treat it as a string, which is wrong).

loicpoulain replied:
Yes, this is what the function is doing here, converting the SSID shell string to a 'byte array', actually to the expected ssid/ssid_length attributes.
I assume the specification defines what is going in the air (SSID is a byte array in 802.11 probe response, beacons...), but how you manage it in software is implementation specific.

GAnthony commented:
> @GAnthony I wonder if this is fixing the issues you had earlier with wifi shell?

Yes, @jukkar, I just pushed a PR to fix this #10863 , seems almost at the same time as this one.
In my case the TI SimpleLink chip silently failed with the wrong SSID, not reporting any callback results.  I thought is was a deeper issue, but turned out to be this new shell quoting behaviour.

You can decide which PR to merge.

Commit:
net: wifi: shell: Fix connect command SSID parameter

The new shell subsystem takes care of removing quote ('"') from
command arguments and produces a standard string.
So now the SSID string starts at &argv[1][0] (instead of &argv[1][1]).
Fix this.

In the same time, remove SSID length parameter which can be find with
strlen().

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

jukkar commented:
> You can decide which PR to merge.

Lets merge #10863 as that one only fixes the quote issue and nothing else. We can ponder this idea whether the SSID length parameter should be removed or not in this PR.

loicpoulain commented:
abandon, since merged with #10863
